x/username/model.go

```diff
@@ -0,0 +1,80 @@
+package username
+
+import (
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+	"github.com/iov-one/weave/orm"
+)
+
+func init() {
+	migration.MustRegister(1, &Token{}, migration.NoModification)
+}
+
+func (ba *BlockchainAddress) Validate() error {
+	switch n := len(ba.BlockchainID); {
+	case n < 3:
+		return errors.Wrap(errors.ErrInput, "blockchain ID too short")
+	case n > 32:
+		return errors.Wrap(errors.ErrInput, "blockchain ID too long")
+	}
+	switch n := len(ba.Address); {
+	case n < 3:
+		return errors.Wrap(errors.ErrInput, "address too short")
+	case n > 1024:
```

Comment:
Woah, that is huuuuuge.
Can you trim it down to 128 bytes? I think 64 is fine, but let's give extra space.

Can you think of any usecase where we need more?

Reply:
I have picked all validation rules without any knowledge what makes a good value. They are all to be refined and declared properly. 
128 bytes sounds great.